bioneuron_oracle/builder.py

```diff
@@ -29,6 +31,22 @@ def __init__(self):
         self.spikes = neuron.h.Vector()
         self.ap_counter.record(neuron.h.ref(self.spikes))
         self.num_spikes_last = 0
+        self._clean = False
+
+    def update(self):
+        if self._clean:
+            raise RuntimeError("cannot update() after cleanup()")
+        count = len(self.spikes) - self.num_spikes_last
+        self.num_spikes_last = len(self.spikes)
+        return count, np.asarray(self.v_record)[-1]
+
+    def cleanup(self):
+        if self._clean:
+            raise RuntimeError("cleanup() may only be called once")
+        self.v_record.play_remove()
+        self.t_record.play_remove()
+        self.spikes.play_remove()
```

Comment:
So something like:
```
def go():
    try:
        with nengo.Simulator(model) as sim:
            sim.run(0.01)
            raise KeyboardInterrupt
    except KeyboardInterrupt:
        pass
    return sim

while True:
    go()
```

Would have leaked memory in the past? Or would it have to be something like:
```
sim1 = go()
sim2 = go()
sim3 = go()
```